Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1

```diff
@@ -186,25 +149,19 @@ else
 
     $connection = Get-VssConnection -TaskContext $distributedTaskContext
 
-    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName" -Verbose
-    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext
+    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter" -Verbose
+    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
     Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName" -Verbose
 
     $fetchedEnvironmentName = $environment.Name
```

Comment:
It's not a blocker for M91. But can we rename this variable that is more intuitive. It is not obvious (that might be cloning) as to why you have to get the name when it was already supplied as input.